### PR TITLE
Logic App local projects

### PR DESCRIPTION
**Pull request by joechung-msft**

### CODE REVIEW

File: package.nls.json [hunk 1]
```diff
@@ -8,7 +8,10 @@
     "azIntegrationAccounts.showExplorerDescription": "Show or hide the Azure Integration Accounts Explorer",
     "azIntegrationAccounts.showSavePrompt": "Prompt the user before saving and updating a Logic App",
     "azIntegrationAccounts.viewProperties": "View properties",
-    "azLogicApps.createLogicApp": "Create",
+    "azLogicApps.addBuildDefinitionToProject": "Add Build Definition to Project...",
```
Comment (refortie) [inline package.nls.json]: None of the other items end in an ellipses, should this be consistent? Aka, should we either remove this one, add it to other ones, or is this a special case?

File: src/commands/logic-app/createProject.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from "fs-extra";
+import * as vscode from "vscode";
+import { ext } from "../../extensionVariables";
+import { openFolder, selectWorkspaceFolder } from "../../utils/workspaceUtils";
+
+export async function createProject(): Promise<void> {
```
Comment (refortie) [inline src/commands/logic-app/createProject.ts]: Should this do something more? When I tested it out, I thought the extension had crashed because it dumped me to an empty workspace (I was already in an empty one, so it didn't look like anything changed). Like should we create a dummy/sample file in the folder to get people started?

Comment (refortie) [reply src/commands/logic-app/createProject.ts]: Perhaps it's just that I'm going into trying it out with no documentation

File: src/commands/logic-app/addLogicAppToProject.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    const options: vscode.ProgressOptions = {
+        location: vscode.ProgressLocation.Notification,
+        title: localize("azLogicApp.addingToProject", "Adding to project...")
+    };
+
+    await vscode.window.withProgress(options, async () => {
+        const logicAppTreeItem = node!.treeItem as LogicAppTreeItem;
+        await logicAppTreeItem.addToProject(workspaceFolderPath);
```
Comment (refortie) [inline src/commands/logic-app/addLogicAppToProject.ts]: After we add a LA to the project, we might want to jump back to the workspace so that we can show the customer that it was added.

Perhaps this could be controlled via a setting if they don't want to jump back after every add. I don't know how many LA people would be adding at a time

File: src/utils/logic-app/templateUtils.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        contentVersion: "1.0.0.0",
+        parameters,
+        resources,
+        variables: {}
+    };
+}
+
+export function generateDeploymentTemplateParameters(parameters: Record<string, any>) {
+    return {
+        $schema: "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
```
Comment (refortie) [inline src/utils/logic-app/templateUtils.ts]: Perhaps pull the strings from this file into constants?

File: package.json [hunk 5]
```diff
@@ -586,15 +623,25 @@
                     "when": "view == azureLogicAppsExplorer"
                 },
                 {
-                    "command": "azureLogicApps.createLogicApp",
+                    "command": "azureLogicApps.createProject",
```
Comment (refortie) [inline package.json]: Should Create Project and Add Build Def to Project be next to each other? Might make sense to group them together, but I also see this as a larger to smaller order, so maybe it's fine.

File: src/wizard/logic-app/GenerateBuildDefinitionStep.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                } else {
+                    resolve(matches);
+                }
+            });
+        });
+        const names = definitions.map((definition) => path.basename(definition).replace(/\.definition\.json$/i, ""));
+
+        for (const name of names) {
+            const file = path.join(workspaceFolderPath!, `${name}.definition.json`);
+            const json = await fse.readJSON(file);
```
Comment (refortie) [inline src/wizard/logic-app/GenerateBuildDefinitionStep.ts]: I haven't tested this out and maybe I've just missed where you handle it. If I modify my JSON files and they aren't valid JSON anymore, what happens?

File: src/commands/logic-app/createProject.ts (hunk 2 above)
Comment (derek1ee) [reply src/commands/logic-app/createProject.ts]: An option could be that we stay on the extension page but give a pop up at lower right with button to "go to workspace"?

File: src/commands/logic-app/addLogicAppToProject.ts (hunk 3 above)
Comment (derek1ee) [reply src/commands/logic-app/addLogicAppToProject.ts]: I'd expect ppl adding more than one at a time so jumping to workspace by default isn't a good idea.

File: package.nls.json (hunk 1 above)
Comment (joechung-msft) [reply package.nls.json]: The rule of thumb is that if it presents a dialog or wizard of some sort, show ellipses.  Otherwise, do not.  In this case, we're showing a wizard.

File: package.json (hunk 5 above)
Comment (joechung-msft) [reply package.json]: The order is consistent with the way Azure Functions presents these options.

File: src/utils/logic-app/templateUtils.ts (hunk 4 above)
Comment (joechung-msft) [reply src/utils/logic-app/templateUtils.ts]: It's only used here for now.  If we start using it elsewhere, e.g., IA deployment templates, we can make a shared constant for it.

File: src/wizard/logic-app/GenerateBuildDefinitionStep.ts (hunk 6 above)
Comment (joechung-msft) [reply src/wizard/logic-app/GenerateBuildDefinitionStep.ts]: It'll error out, e.g., "Unexpected end of JSON input," and terminate the process.  The wizard will catch the rejected promise and display the message.

File: src/commands/logic-app/createProject.ts (hunk 2 above)
Comment (joechung-msft) [reply src/commands/logic-app/createProject.ts]: I will add a progress notification for when this completes.
